fabric-input-events-v0/src/main/java/net/fabricmc/fabric/api/client/FabricMouse.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client;
+
+import net.fabricmc.fabric.impl.client.FabricMouseImpl;
+
+public interface FabricMouse {
+	public static FabricMouse INSTANCE = FabricMouseImpl.INSTANCE;
+
+	public double impl_getX();
+	public double impl_getY();
+	public int impl_getPressedButtons();
+	public boolean impl_isButtonPressed(int button);
+	public int impl_getMods();
```

Comment:
These all can be moved into the `impl` package and accessed via util methods there